### PR TITLE
Floyd Warshall Algorithm in php 

### PR DESCRIPTION
**Pull request by mharshita**
Fixes #2205 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- added Floyd Warshall Algorithm implementation in php

#### Languages Used:
- PHP

#### Files Added:
- Floyd_Warshall_Algorithm.php

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by mharshita** [comment]
@jainaman224  @somya-kapoor  @asha15  Please review my PR. I have added implementation of Floyd Warshall Algorithm in PHP.

File: Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
+<?php
+
+//Floyd-Warshall Algorithm is an algorithm for finding the shortest path between all the pairs of vertices in a weighted graph.
+
+$f    = fopen('php://stdin', 'r');
```
Comment (asha15) [inline Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php:5]: please remove the extra space before equal sign

File: Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+$line = fgets($f);
+
+$arr  = explode(" ", $line);
```
Comment (asha15) [inline Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php:9]: please remove the extra space before equal sign

File: Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+$m    = $arr[0]; // first int as m rows 
+$n    = $arr[1]; // second int as n cols
+
+unset($arr[0]);
+unset($arr[1]);
+
+$arr    = array_values($arr);
```
Comment (asha15) [inline Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php:17]: please remove the extra space before equal sign

File: Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+                echo str_pad("INF", 7);
+            }
+            else
+            {
+                echo str_pad($distance[$i][$j], 7);
+            }
+		}
+		echo "<br/>";
+	}
+	echo  "</pre>";
```
Comment (asha15) [inline Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php:53]: please correct the indentation

File: Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+function FloydWarshall($graph, $verticesCount)
+{
+	$distance = array();
```
Comment (asha15) [inline Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php:58]: correct the indentation

File: Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+		for ($i = 0; $i < $verticesCount; ++$i)
+		{
+			for ($j = 0; $j < $verticesCount; ++$j)
+			{
+                if ($distance[$i][$k] + $distance[$k][$j] < $distance[$i][$j])
+                {
+                    $distance[$i][$j] = $distance[$i][$k] + $distance[$k][$j];
+                }
+			}
+		}
```
Comment (asha15) [inline Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php:79]: correct the indentation

File: Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php (hunk 1 above)
Comment (mharshita) [reply Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php]: @asha15 Done the required changes. Please review.

File: Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php (hunk 2 above)
Comment (mharshita) [reply Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php]: @asha15  Done the required changes. Please review.

File: Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php (hunk 3 above)
Comment (mharshita) [reply Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php]: @asha15  Done the required changes. Please review.

File: Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php (hunk 4 above)
Comment (mharshita) [reply Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php]: @asha15  Done the required changes. Please review.

File: Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php (hunk 5 above)
Comment (mharshita) [reply Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php:61]: @asha15  Done the required changes. Please review.

File: Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php (hunk 6 above)
Comment (mharshita) [reply Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php]: @asha15  Done the required changes. Please review.

**Comment by mharshita** [comment]
@asha15  I have made the requested changes and squashed the commits. Please review.

**Comment by mharshita** [comment]
@asha15  Sorry for the inconvenience. I have corrected the indentation. Please review.

File: Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+             if ($distance[$i][$j] == $INF)
+             {
+                 echo str_pad("INF", 7);
+             }
+             else
+             {
+                echo str_pad($distance[$i][$j], 7);
+             }
+	}
+	echo "<br/>";
```
Comment (asha15) [inline Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php:51]: please correct the indentation in curly braces and echo statement

File: Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+        for ($i = 0; $i < $verticesCount; ++$i)
+	{
+	    for ($j = 0; $j < $verticesCount; ++$j)
+	    {
+                if ($distance[$i][$k] + $distance[$k][$j] < $distance[$i][$j])
+                {
+                    $distance[$i][$j] = $distance[$i][$k] + $distance[$k][$j];
+                }
+	    }
+	}
```
Comment (asha15) [inline Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php:79]: here too

File: Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php (hunk 7 above)
Comment (mharshita) [reply Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php]: @asha15  Done. Please review.

File: Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php (hunk 8 above)
Comment (mharshita) [reply Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php]: @asha15  Done. Please review.

**Comment by mharshita** [comment]
@asha15 I have made the required changes and squashed the commits. Please review.

File: Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+         for ($j = 0; $j < $verticesCount; ++$j)
+	 {
+             if ($distance[$i][$j] == $INF)
+                 echo str_pad("INF", 7);
+             else
+                echo str_pad($distance[$i][$j], 7);
+	}
+	echo "<br/>";
+    }
+    echo  "</pre>";
```
Comment (asha15) [inline Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php:49]: correct the indentation

File: Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+    for ($k = 0; $k < $verticesCount; ++$k)
+    {
+        for ($i = 0; $i < $verticesCount; ++$i)
+	{
+	    for ($j = 0; $j < $verticesCount; ++$j)
+	    {
+                if ($distance[$i][$k] + $distance[$k][$j] < $distance[$i][$j])
+                    $distance[$i][$j] = $distance[$i][$k] + $distance[$k][$j];
+	    }
+	}
```
Comment (asha15) [inline Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php:73]: correct the indentation

**Comment by mharshita** [comment]
@asha15  It is showing correct indentation on editing page.

**Comment by somya-kapoor** [comment]
Change the indent type to space and indent size to 4 in edit mode.

**Comment by mharshita** [comment]
@somya-kapoor  Done that.

**Comment by mharshita** [comment]
@somya-kapoor  In the edit window, it is showing correct indentation

File: Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+
+function PrintResult($distance, $verticesCount)
+{
+    global $INF;	
+    echo "<pre>" . "Shortest distances between every pair of vertices:" . "<br/>";
+
+    for ($i = 0; $i < $verticesCount; ++$i)
+    {
+         for ($j = 0; $j < $verticesCount; ++$j)
+	 {
```
Comment (asha15) [inline Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php:41]: see you didn't indent the open curly braces for the for statement here

**Comment by mharshita** [comment]
@asha15  In the edit window it is showing correct indentation.
image

File: Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+    for ($k = 0; $k < $verticesCount; ++$k)
+    {
+        for ($i = 0; $i < $verticesCount; ++$i)
+	    {
+	        for ($j = 0; $j < $verticesCount; ++$j)
+	        {
+                if ($distance[$i][$k] + $distance[$k][$j] < $distance[$i][$j])
+                    $distance[$i][$j] = $distance[$i][$k] + $distance[$k][$j];
+	        }
+	    }
```
Comment (asha15) [inline Floyd_Warshall_Algorithm/Floyd_Warshall_Algorithm.php:73]: still have some mistakes

**Pull request closed without merging**